shared/reducers/wallets.tsx

```diff
@@ -243,6 +243,48 @@ export default function(state: Types.State = initialState, action: WalletsGen.Ac
         builtPayment: state.get('builtPayment').merge({toErrMsg: ''}),
         builtRequest: state.get('builtRequest').merge({toErrMsg: ''}),
       })
+    case WalletsGen.clearBuildingAdvanced:
+      return state.set('buildingAdvanced', Constants.emptyBuildingAdvanced)
+    case WalletsGen.setBuildingAdvancedRecipient:
+      return state.update('buildingAdvanced', buildingAdvanced =>
+        buildingAdvanced.set('recipient', action.payload.recipient)
+      )
+    case WalletsGen.setBuildingAdvancedRecipientAmount:
+      return state
+        .update('buildingAdvanced', buildingAdvanced =>
+          buildingAdvanced.set('recipientAmount', action.payload.recipientAmount)
+        )
+        .set('builtPaymentAdvanced', Constants.emptyBuiltPaymentAdvanced)
+    case WalletsGen.setBuildingAdvancedRecipientAsset:
+      return state
+        .update('buildingAdvanced', buildingAdvanced =>
+          buildingAdvanced.set('recipientAsset', action.payload.recipientAsset)
+        )
+        .set('builtPaymentAdvanced', Constants.emptyBuiltPaymentAdvanced)
+    case WalletsGen.setBuildingAdvancedRecipientType:
+      return state.update('buildingAdvanced', buildingAdvanced =>
+        buildingAdvanced.set('recipientType', action.payload.recipientType)
+      )
+    case WalletsGen.setBuildingAdvancedPublicMemo:
+      return state.update('buildingAdvanced', buildingAdvanced =>
+        buildingAdvanced.set('publicMemo', action.payload.publicMemo)
+      )
+    case WalletsGen.setBuildingAdvancedSenderAccountID:
+      return state.update('buildingAdvanced', buildingAdvanced =>
+        buildingAdvanced.set('senderAccountID', action.payload.senderAccountID)
+      )
+    case WalletsGen.setBuildingAdvancedSenderAsset:
+      return state
+        .update('buildingAdvanced', buildingAdvanced =>
+          buildingAdvanced.set('senderAsset', action.payload.senderAsset)
+        )
+        .set('builtPaymentAdvanced', Constants.emptyBuiltPaymentAdvanced)
+    case WalletsGen.setBuildingAdvancedSecretNote:
+      return state.update(
+        'buildingAdvanced',
+        buildingAdvanced => buildingAdvanced.set('secretNote', action.payload.secretNote)
+        // TODO clear error when we have that
```

Comment:
instead of a bare TODO, mention whether it's to be fixed pre-merge (perhaps with a checklist item in the PR desc) or post-merge (with a ticket ID)